### PR TITLE
Wait for nova api service after reboot

### PR DESCRIPTION
**Pull request by toabctl**
After a controller reboot, "nova list" fails with:

ERROR (ConnectionRefused): Unable to establish connection to \
      http://192.168.242.2:8774/v2/my-uuid/servers/detail

Wait for the nova api service before doing the "nova list"

### CODE REVIEW

File: scripts/qa_crowbarsetup.sh [hunk 1]
```diff
@@ -2908,6 +2908,18 @@ function onneutron_wait_for_neutron()
     fi
 }
 
+function get_nova_multi_controller_node()
```
Comment (aspiers) [inline scripts/qa_crowbarsetup.sh]: -1, this simply duplicates `get_novacontroller` (and does it wrong - it will fail in HA mode, see #477)

Comment (toabctl) [inline scripts/qa_crowbarsetup.sh]: @aspiers Done in https://github.com/toabctl/automation/commit/4bf09bd8231e92f7f0c249a4473c45ff344c9e58 .

**Comment by aspiers** [comment]
Thanks, +1 in principle although please can you wait until #477 is merged and then change this to reference `novacontrollersvc` so that it works with HA too.

**Comment by toabctl** [comment]
> Thanks, +1 in principle although please can you wait until #477 is merged and then change this to reference novacontrollersvc so that it works with HA too.

This one is hopefully fixing the current jenkins failures we need for M2. So imo should be merged first...

File: scripts/qa_crowbarsetup.sh [hunk 2]
```diff
@@ -2908,6 +2908,12 @@ function onneutron_wait_for_neutron()
     fi
 }
 
+function onnovacontroller_wait_for_nova()
+{
+    get_novacontroller
+    wait_for 300 3 "ssh $novacontroller 'service openstack-nova-api status'" "nova-api service running state"
```
Comment (dirkmueller) [inline scripts/qa_crowbarsetup.sh]: I guess you also want to wait for nova-manage service list being sucessful, see testsetup function. perhaps we can use that same code here as well (e.g. putting it in a common function)

**Comment by aspiers** [comment]
> This one is hopefully fixing the current jenkins failures we need for M2. So imo should be merged first...

They are both fixing jenkins failures, but if you merge this first, it will need to be changed immediately when #477 is merged ...

File: scripts/qa_crowbarsetup.sh (hunk 2 above)
Comment (toabctl) [inline scripts/qa_crowbarsetup.sh]: @dirkmueller Done in https://github.com/SUSE-Cloud/automation/commit/7d72aafd9086910cc6951bdf57e573f7a69534fd

**Comment by toabctl** [comment]
Seems that this "fix" is not needed anymore...

**Pull request closed without merging**